Assets/Scripts/Data/Core/RandomBuffer.cs

```diff
@@ -19,34 +19,42 @@ protected virtual IList<T> source
 	}
 
 	T[] backingSource;
-	Random random;
+	protected Random random;
 	// Using a hash set for fast lookup
 	HashSet<int> usedIndices;
```

Comment:
Might want to reorder these variables. I think we agreed to sort my protection level.